src/Auth0Client.ts

```diff
@@ -1106,7 +1115,8 @@ export default class Auth0Client {
           grant_type: 'authorization_code',
           redirect_uri: params.redirect_uri,
           auth0Client: this.options.auth0Client,
-          useFormData: this.options.useFormData
+          useFormData: this.options.useFormData,
+          timeout: this.httpTimeoutMs
```

Comment:
actually, this is the place where I thought we might break `getTokenSilently({ timeout: ... })` - not sure if anyone is using that?

Maybe we should do: `timeout: customOptions.timeout || this.httpTimeoutMs`?

Reply:
Ok, stepping back a bit the API does look a bit odd here.

With this PR it's now possible to construct a config like this:

```js
const client = new Auth0Client({
  domain,
  clientId,
  httpTimeoutInSeconds: 60,
  authorizeTimeoutInSeconds: 60
});

// Then call getTokenSilently like this.
// `timeoutInSeconds` only affects the timeout on /authorize but no way to configure http timeout here
await client.getTokenSilently({ timeoutInSeconds: 30 });
```

Should we also add this config in to override at the `getTokenSilently` level, or do you think just explaining the behaviour of `timeoutInSeconds` in the docs is enough?

Reply:
Regardless, this:

```js
Maybe we should do: timeout: customOptions.timeout || this.httpTimeoutMs?
```

`customOptions.timeout` here is documented as being "A maximum number of seconds to wait before declaring the background /authorize call as failed for timeout", so the two timeouts relate to different things. But would be confusing as I mentioned above.

Reply:
`customOptions.timeout` isn't documented anywhere, that's the docs for `timeoutInSeconds`

Reply:
Currently, if I call `auth0.getTokenSilently({ timeout: 100 })` the HTTP Timeout for the code exchange will be 100ms

Reply:
Right, sorry - I think I see what you're saying. You're covering the case where someone has inadvertently used `getTokenSilently({ timeout: 100 });` instead of the documented `getTokenSilently({ timeoutInSeconds: 100 });`, which they could well do because we allow arbitrary parameters?

It would be easy to cover off that use case, sure - just in case.

Reply:
I've changed this now 👍🏻 was a good spot, thanks